File: frontend/src/types/cloakAnnotation.ts
// Types pour la couche d'annotation CLOAK — séparée du contenu source.
// ⚠️ Le framework CLOAK original est en lecture seule ; seules ces annotations sont éditables.

export type AnnotationStatus = 'a_tester' | 'vu_en_lab' | 'maitrise' | ''

export interface CloakAnnotation {
  id: number
  technique_ref: string   // ex: "tac-3:te-9" ou "tac-3:te-9:sub-2"
  user_notes: string      // markdown libre
  status: AnnotationStatus
  counter_notes: string   // contre-mesures personnelles (markdown)
  tags: string            // CSV ou JSON array
  created_at: string
  updated_at: string
}

export interface CloakAnnotationRequest {
  technique_ref: string
  user_notes: string
  status: AnnotationStatus
  counter_notes: string
  tags: string
}

/** Map technique_ref → annotation pour accès O(1) dans CLOAKPage */
export type AnnotationMap = Map<string, CloakAnnotation>

File: backend/internal/models/cloak_annotation.go
package models

import "time"

// CloakAnnotation — couche d'annotation personnelle au-dessus du framework CLOAK.
// ⚠️ Le contenu source CLOAK reste intact et immuable.
// Seule cette table est modifiée ; un reset supprime la ligne, pas la fiche CLOAK.
type CloakAnnotation struct {
	ID           uint      `json:"id"            gorm:"primaryKey"`
	TechniqueRef string    `json:"technique_ref" gorm:"not null;uniqueIndex"` // ex: "tac-3:te-9" ou "tac-3:te-9:sub-2"
	UserNotes    string    `json:"user_notes"`    // markdown libre (observations, commandes testées…)
	Status       string    `json:"status"`        // "a_tester" | "vu_en_lab" | "maitrise" | ""
	CounterNotes string    `json:"counter_notes"` // contre-mesures personnelles (règles Sigma, SIEM…)
	Tags         string    `json:"tags"`          // JSON array ou CSV libre
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

type CloakAnnotationRequest struct {
	TechniqueRef string `json:"technique_ref" binding:"required"`
	UserNotes    string `json:"user_notes"`
	Status       string `json:"status"`
	CounterNotes string `json:"counter_notes"`
	Tags         string `json:"tags"`
}
